src/docs/sphinx/dev_guide/updating_tpls.rst

```diff
@@ -0,0 +1,221 @@
+---------------------------------------------
+
+The following instructions describe how to install local copies of Axom
+TPLs on Livermore Computing (LC) platforms and recreate our Docker containers
+with a new set of TPLs. Typically, this process is followed when you want to 
+update one or more TPLs which Axom depends on. After they are built and
+the required changes are merged into develop, they will be available for
+other Axom developers to use during development, in Axom Gitlab CI testing, etc.
+
+#. **Working on a local branch.** 
+   Make a local clone of the Axom repo and create a branch to work on. Working
+   on a separate branch is a good practice to avoid propagating mis-steps
+   to other users and/or developers.
+
+#. **Changing versions of system packages or other TPLs.**
+   To change a version of a system package, which applies to an LC platforms 
+   or a Docker container image we use for CI testing on GitHub, go into
+   the directory ``axom/scripts/spack/configs``. There you will find a 
+   sub-directory for each supported LC system type. Each sub-directory
+   has a ``packages.yaml`` file which contains an entry for each system level
+   package we rely on. Find the entry for the library you wish to update and 
+   change the version number. Do this for each system you want to test/change,
+   including configurations in the `docker` subdirectory.
+
+   .. note:: Alongside each ``packages.yaml`` in each system package directory,
+             there is a ``compilers.yaml`` file containing compiler and 
+             version information for compilers we use for development and 
+             testing. If you wish to test and build with a new compiler or 
+             version on a system, modify the appropriate ``compilers.yaml`` 
+             file.
+
+   To change a version of a non-system TPL, go into the 
+   ``axom/scripts/spack/packages`` directory. There you will find a 
+   sub-directory for each TPL Axom uses. Modify the contents of the Spack
+   package file ``package.py`` in each package sub-directory as needed. 
+
+   .. note:: Before continuing, you will want to test the installation works
+             on all LC systems with the steps in :ref:`local-tpls-label`.
+
+
+#. **Install TPLs on all required LC machines.**
+   This step need to be run on each of the machines named in Axom's standard host-configs.
+   When you are confident that everything is correct, become the service user
+   ``atk`` via the following command::
+
+   $ xsu atk
+
+   .. note:: This command requires a certain level of permission.
+
+   Run the corresponding command for the system you are on::
+
+     # blueos
+     $ lalloc 1 -W 120 scripts/llnl/build_tpl.py
+     
+     # toss3
+     $ srun -N1 --interactive -t 120 scripts/llnl/build_tpl.py
+
+   This script will build all third-party libraries for all compilers specs
+   for the machine you are on. These will be installed into shared directories
```

Comment:
A couple of questions about this:

1. Should we note the LC directory where these are installed?
2. What happens if a build fails? Do broken build artifacts get installed, or deleted? Do all the other builds generated by the script also get delted, even if successful?

Reply:
i fixed 1.. and added a note about when you should cleanup old builds.  To answer the last part, no we do not delete old builds as a part of any script.  we generally wait a long time then bulk delete them manually.